number_parser/parser.py

```diff
@@ -1,65 +1,67 @@
 import re
 import string
+import os
+import json
 # The initial global dictionaries that have been declared below will be changed,
 # should be easily derived automatically from CLDR data/other source for each of the languages.
 
 # We will need to have all values from 1-100 in a dictionary for some languages like hindi while others
 # english / french build 2 digit numbers using MTEN dictionary values.
 # (Difference observable in hi.json and en.json in numeral_language_data)
 
-MULTIPLIERS = {
-    "thousand": 1000,
-    "thousands": 1000,
-    "million": 1000000,
-    "millions": 1000000,
-    "billion": 1000000000,
-    "billions": 1000000000,
-    "trillion": 1000000000000,
-    "trillions": 1000000000000,
-}
-
-VALID_TOKENS_IN_NUMBERS = ["and", "-"]
-
-UNITS = {
-    word: value
-    for value, word in enumerate(
-        "one two three four five six seven eight nine".split(), 1
-    )
-}
-
-STENS = {
-    word: value
-    for value, word in enumerate(
-        "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split(), 10
-    )
-}
-
-MTENS = {
-    word: value * 10
-    for value, word in enumerate(
-        "twenty thirty forty fifty sixty seventy eighty ninety".split(), 2
-    )
-}
-
-HUNDRED = {"hundred": 100, "hundreds": 100}
-
-ALL_WORDS = {**UNITS, **STENS, **MTENS, **HUNDRED, **MULTIPLIERS}
-BASE_NUMBERS = {**UNITS, **STENS}
+LANGUAGE_DIRECTORY = "../number_parser/translation_data_merged/"
+UNIT_NUMBERS = {}
+BASE_NUMBERS = {}
+MTENS = {}
+MHUNDREDS = {}
+MULTIPLIERS = {}
+VALID_TOKENS_IN_NUMBERS = []
+ALL_WORDS = {}
+ALL_BASE_WORDS = {}
+
+def populate_dictionaries(lang):
+    file_name = lang + ".json"
+    fpath = os.path.join(LANGUAGE_DIRECTORY, file_name)
+
+    with open(fpath, 'r') as lang_file:
+        lang_data = json.load(lang_file)
+        unit_numbers = lang_data["UNIT_NUMBERS"]
+        base_numbers = lang_data["BASE_NUMBERS"]
+        mtens = lang_data["MTENS"]
+        mhundreds = lang_data["MHUNDREDS"]
+        multipliers = lang_data["MULTIPLIERS"]
+        valid_tokens_in_numbers = lang_data["VALID_TOKENS"]["tokens"]
+        all_words = {**unit_numbers, **base_numbers, **mtens, **mhundreds, **multipliers}
+        all_base_words = {**unit_numbers, **base_numbers}
+
+        UNIT_NUMBERS.update(unit_numbers)
+        BASE_NUMBERS.update(base_numbers)
+        MTENS.update(mtens)
+        MHUNDREDS.update(mhundreds)
+        MULTIPLIERS.update(multipliers)
+        ALL_WORDS.update(all_words)
+        ALL_BASE_WORDS.update(all_base_words)
+        VALID_TOKENS_IN_NUMBERS[:] = valid_tokens_in_numbers
 
 def check_validity(current_token, previous_token):
```

Comment:
I think it is a good idea for improving the interface to start this function name with an "underscore" (`_`) to mark it as a private function. This is common in Python to indicate that this function is private and it could change or even disappear in the future, so it's not recommended to use it from outside this module/file.

the result would be `_check_validity()`. The same for the `populate_dictionaries()` function (or the new name it will get).